### DISCUSSION TITLE
How to call several .ini files correctly with "-c" argument?

### DISCUSSION THREAD

**Discussion by Coder2023Coder** [pytest-dev/pytest#10861] [Q&A] [answered]
I am designing a testing framework based on using `pytest`.

`pytest` will be called in a python script and not from the command line.

I have several directories in each of which there are:

 - a file called `tests.py` in which `test_` functions are placed.
 - a `.ini` file

What I am trying to do is to store each testing report in its corresponding directory. That is the why I am using a separate `.ini` file in each directory.

Following is the content of each `.ini` file:
```
[pytest]
addopts = --html=<path>/test_report.html
```

And the whole setup is:
```
root
+-- dir_1
|   +-- d1.ini
|   +-- tests_d1.py
+-- dir_2
|   +-- d2.ini
|   +-- tests_d2.py
.
.
.
+-- dir_n
|   +-- dn.ini
|   +-- tests_dn.py
```

When calling `pytest.main`, I do it as follows:

```
pytest_params = ['-v', <path> + tests.py,
                 '-c', <path> + <some name>.ini]

pytest.main(pytest_params)
```

But when I run `pytest.main` I get the following error:

```
ERROR: usage: main.py [options] [file_or_dir] [file_or_dir] [...]
main.py: error: unrecognized arguments: --html=<path>/test_report.html <path>/tests.py 
  inifile: <path>/<some name>.ini
  rootdir: <path>
```
(I removed the paths from the error message). 

How do I fix the error?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
at first glance this looks like pytest-html is not installed, please instal it and/or provide more details on your environment

**Reply by Coder2023Coder** [reply]
Thank you. I installed it but now I get:
```
import file mismatch:
imported module 'tests' has this __file__ attribute:
  /home/project/A/tests.py
which is not the same as the test file we want to collect:
  /home/project/B/tests.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
```

**Reply by RonnyPfannschmidt** [reply]
`__init__.py` files need to be added or the importlib mode needs to be used

**Reply by Coder2023Coder** [reply]
Amazing. It works now. Thank you very much.
